karpenter-operator/controllers/nodeclass/ec2_nodeclass_controller.go

```diff
@@ -0,0 +1,394 @@
+		InstanceStorePolicy:      openshiftEC2NodeClass.Spec.KarpenterInstanceStorePolicy(),
+		MetadataOptions:          openshiftEC2NodeClass.Spec.KarpenterMetadataOptions(),
+		Context:                  openshiftEC2NodeClass.Spec.Context,
+	}
+
+	var subnetSelectorTerms []awskarpenterv1.SubnetSelectorTerm
+	if openshiftEC2NodeClass.Spec.SubnetSelectorTerms != nil {
+		for _, term := range openshiftEC2NodeClass.Spec.SubnetSelectorTerms {
+			subnetSelectorTerms = append(subnetSelectorTerms, awskarpenterv1.SubnetSelectorTerm{
+				Tags: term.Tags,
+				ID:   term.ID,
+			})
+		}
+	} else {
+		subnetSelectorTerms = []awskarpenterv1.SubnetSelectorTerm{
+			{
+				Tags: map[string]string{
+					"karpenter.sh/discovery": hcp.Spec.InfraID,
+				},
+			},
+		}
+	}
+	ec2NodeClass.Spec.SubnetSelectorTerms = subnetSelectorTerms
+
+	var securityGroupSelectorTerms []awskarpenterv1.SecurityGroupSelectorTerm
+	if openshiftEC2NodeClass.Spec.SecurityGroupSelectorTerms != nil {
+		for _, term := range openshiftEC2NodeClass.Spec.SecurityGroupSelectorTerms {
+			securityGroupSelectorTerms = append(securityGroupSelectorTerms, awskarpenterv1.SecurityGroupSelectorTerm{
+				Tags: term.Tags,
+				ID:   term.ID,
+				Name: term.Name,
+			})
+		}
+	} else {
+		securityGroupSelectorTerms = []awskarpenterv1.SecurityGroupSelectorTerm{
+			{
+				Tags: map[string]string{
+					"karpenter.sh/discovery": hcp.Spec.InfraID,
+				},
+			},
+		}
+	}
+	ec2NodeClass.Spec.SecurityGroupSelectorTerms = securityGroupSelectorTerms
+
+	return nil
+}
+
+func (r *EC2NodeClassReconciler) reconcileStatus(ctx context.Context, ec2NodeClass *awskarpenterv1.EC2NodeClass, openshiftNodeClass *hyperkarpenterv1.OpenshiftEC2NodeClass) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	originalObj := openshiftNodeClass.DeepCopy()
+
+	openshiftNodeClass.Status = hyperkarpenterv1.OpenshiftEC2NodeClassStatus{}
+	for _, securityGroup := range ec2NodeClass.Status.SecurityGroups {
+		openshiftNodeClass.Status.SecurityGroups = append(openshiftNodeClass.Status.SecurityGroups, hyperkarpenterv1.SecurityGroup{
+			ID:   securityGroup.ID,
+			Name: securityGroup.Name,
+		})
+	}
+	for _, subnet := range ec2NodeClass.Status.Subnets {
+		openshiftNodeClass.Status.Subnets = append(openshiftNodeClass.Status.Subnets, hyperkarpenterv1.Subnet{
+			ID:     subnet.ID,
+			Zone:   subnet.Zone,
+			ZoneID: subnet.ZoneID,
+		})
+	}
+	for _, condition := range ec2NodeClass.Status.Conditions {
+		openshiftNodeClass.Status.Conditions = append(openshiftNodeClass.Status.Conditions, metav1.Condition(condition))
+	}
+
+	if !reflect.DeepEqual(originalObj.Status, openshiftNodeClass.Status) {
+		if err := r.guestClient.Status().Patch(ctx, openshiftNodeClass, client.MergeFrom(originalObj)); err != nil {
+			return fmt.Errorf("failed to update status: %v", err)
+		}
+	}
+
+	log.Info("Reconciled OpenshiftEC2NodeClass status")
+	return nil
+}
+
+func (r *EC2NodeClassReconciler) reconcileVAP(ctx context.Context) error {
```

Comment:
what happens if the ec2class is deleted out of band?

Reply:
we will just create it back. should we prevent deletion as well?